src/main/java/org/jenkinsci/backend/ircbot/IrcListener.java

```diff
@@ -578,6 +604,89 @@ private boolean createComponent(Channel channel, User sender, String subcomponen
+                    return null;
+                }
+
+                try {
+                    repo.getBranch(branchName);
+                } catch (IOException e) {
+                    repo.createRef("refs/heads/" + branchName, repo.getBranch("master").getSHA1());
+                }
+
+                GHContentBuilder builder = repo.createContent();
+                builder = builder.branch(branchName).message("Adding upload permissions file for " + forkTo);
+                String name = forkTo.replace("-plugin", "");
+                String content = "---\n" +
+                        "name: \"" + name + "\"\n" +
+                        "github: \"jenkinsci/" + forkTo + "\"\n" +
+                        "paths:\n" +
+                        "- \"" + artifactPath + "\"\n" +
+                        "developers:\n";
+                for(String u : releaseUsers) {
+                    content += "- \"" + u + "\"\n";
+                }
+
+                builder = builder.content(content).path("permissions/plugin-" + forkTo.replace("-plugin", "") + ".yml");
+
+                GHContentUpdateResponse res = builder.commit();
+
+                Map<String, String> replacements = new HashMap<>();
+                String description = String.format("https://github.com/%s/%s\n%s/browse/%s", IrcBotConfig.GITHUB_ORGANIZATION, forkTo, IrcBotConfig.JIRA_URL, jiraIssue);
+
+                replacements.put("<!-- fill in description here, this will at least be a link to a GitHub repository, and often also links to hosting request, and @mentioning other committers/maintainers as per the checklist below -->", description);
+                replacements.put("- [ ] Add link to plugin/component Git repository in description above", "- [x] Add link to plugin/component Git repository in description above");
+                replacements.put("- [ ] Add link to resolved HOSTING issue in description above", "- [x] Add link to resolved HOSTING issue in description above");
+                replacements.put("- [ ] Make sure the file is created in `permissions/` directory", "- [x] Make sure the file is created in `permissions/` directory");
+                replacements.put("- [ ] `artifactId` (pom.xml) is used for `name` (permissions YAML file).", "- [x] `artifactId` (pom.xml) is used for `name` (permissions YAML file).");
+                replacements.put("- [ ] [`groupId` / `artifactId` (pom.xml) are correctly represented in `path` (permissions YAML file)](https://github.com/jenkins-infra/repository-permissions-updater/#managing-permissions)", "- [x] [`groupId` / `artifactId` (pom.xml) are correctly represented in `path` (permissions YAML file)](https://github.com/jenkins-infra/repository-permissions-updater/#managing-permissions)");
+                replacements.put("- [ ] Check that the file is named `plugin-${artifactId}.yml` for plugins", "- [x] Check that the file is named `plugin-${artifactId}.yml` for plugins");
```

Comment:
This seems unnecessarily brittle given it depends on the exact text of the template.

Can we just replace the entire text and use special content for bot-submitted PRs?

Reply:
Sure. I thought of that, but didn't know how people would feel. I think having the bot put in only the links for the repo and hosting issue would suffice. Does that sound ok to you?

Reply:
Done, example PR https://github.com/slide-testing/repository-permissions-updater/pull/7

Reply:
Nice, if a bit minimalist, especially given that _something_ can go wrong and we should have a reminder for it to help those processing the PR.

I'd replace the entire text with the following:

```
Hello from your friendly Jenkins Hosting Bot!

This is an automatically created PR for:

- https://issues.jenkins.io/browse/HOSTING-12345
- https://github.com/jenkinsci/strawberry-cheesecake-plugin

The user listed this PR may not have logged in to Artifactory yet, check the PR status.
To check again, close and reopen this PR.

Reply:
less notifications for others if you just re-trigger via checks tab? the user who this is created for won't be able to re-trigger it, needs to be the hosting team (since the user isn't creating the PR)

Reply:
> less notifications for others if you just re-trigger via checks tab?

Ah OK, didn't know that worked 👍

Reply:
Now I really want a strawberry-cheesecake plugin :-) I'll update the text.

Reply:
Also, I think the user won't be able to close and reopen, so it would have to be the hosting team anyway that did that too.